### DISCUSSION TITLE
Node Sync stuck in re-org loop

### DISCUSSION THREAD

**Discussion by zkharkhach2** [zkSync-Community-Hub/zksync-developers#668] [General]
### Environment

Mainnet

### L2 block number

13098280

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Hello all, we currently run 24.12.0, but this is a problem with previous versions as well. The note hits the following loop:

```zknode  | 2024-06-04T07:23:45.999512Z  INFO zksync_state::postgres: Storage values cache is too far behind (current L2 block is 0; requested update to 13098175); resetting the cache
zknode  | thread 'tokio-runtime-worker' panicked at core/lib/types/src/fee_model.rs:60:18:
zknode  | Can not convert PubdataIndependentBatchFeeModelInput into L1PeggedBatchFeeModelInput
zknode  | stack backtrace:
zknode  |    0: rust_begin_unwind
zknode  |              at ./rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/std/src/panicking.rs:617:5
zknode  |    1: core::panicking::panic_fmt
zknode  |              at ./rustc/5c6a7e71cd66705c31c9af94077901a220f0870c/library/core/src/panicking.rs:67:14
zknode  |    2: zksync_types::fee_model::BatchFeeInput::into_l1_pegged
zknode  |    3: multivm::glue::types::vm::block_context_mode::<impl multivm::glue::GlueFrom<multivm::interface::types::inputs::l1_batch_env::L1BatchEnv> for multivm::versions::vm_m5::vm_with_bootloader::BlockContextMode>::glue_from
zknode  |    4: <multivm::versions::vm_1_3_2::vm::Vm<S,H> as multivm::interface::traits::vm::VmInterface<S,H>>::new
zknode  |    5: multivm::vm_instance::VmInstance<S,H>::new_with_specific_version
zknode  |    6: zksync_core::api_server::execution_sandbox::apply::Sandbox::into_vm
zknode  |    7: zksync_core::api_server::execution_sandbox::execute::TransactionExecutor::execute_tx_in_sandbox::{{closure}}::{{closure}}::{{closure}}
zknode  |    8: tokio::runtime::task::core::Core<T,S>::poll
zknode  |    9: tokio::runtime::task::harness::Harness<T,S>::poll

```zknode  | 2024-06-04T07:25:40.647967Z  INFO zksync_storage::db: All the RocksDB instances are dropped
zknode  | 2024-06-04T07:25:40.648048Z ERROR zksync_utils::wait_for_tasks: One of actors returned an error during shutdown: reorg_detector.run()
zknode  |
zknode  | Caused by:
zknode  |     reorg detected, restart the node to revert to the last correct L1 batch #196685.

### Repo Link (Optional)

_No response_

### Additional Details

_No response_

**Comment by alhenaai** [comment]
It appears that your ZKsync node is encountering a reorganization (reorg) loop, which is causing it to restart repeatedly. This issue typically indicates a problem with the node's ability to synchronize with the correct L1 batch data or a bug in the node's implementation.

Here are a few steps and considerations to help resolve the issue:

1. **Check for Updates**: Ensure that your node is running the latest version of the software. Sometimes, bugs that cause such issues are resolved in newer releases.

2. **Configuration Check**: Verify that your node's configuration settings are correct, especially those related to L1 connection and batch processing.

3. **Logs and Errors**: Review the node logs for any additional error messages or warnings that could provide more insight into what might be causing the reorg loop.

4. **Restart with Clean State**: Consider restarting the node with a clean state. This might involve clearing cache or resetting database states that could be causing inconsistencies.

5. **Increase Resources**: If the issue persists, it might be related to resource constraints. Ensure that your node has sufficient memory and storage, as a lack of resources can lead to such issues.

6. **Community and Support**: If the problem continues after trying the above steps, consider reaching out to the ZKsync community or support for more specialized assistance. They might have encountered similar issues and could offer a solution.

For detailed troubleshooting steps and more information on handling node issues, you can refer to the [ZKsync Node Troubleshooting Documentation](https://docs.zksync.io/zksync-node/troubleshooting).

If these steps do not resolve the issue, it might be beneficial to provide more detailed logs and configuration details to the support or community forums for further analysis.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply]
@zkharkhach2 did the above help you? If not could you downvote so the bot can learn?
